### DISCUSSION TITLE
DAT: Github Footer

### DISCUSSION THREAD

**Discussion by emmatseng-ux** [gc-da11yn/gc-da11yn.github.io#305] [Content (Open Discussion)]
Hi everyone! We are currently revisiting the words used in the footer and taking into consideration the Flesch Kincaid reading level, use of plain language and inclusive language, and contextual use of the items in the footer (note: the target audience for source code is mainly used by developers but anyone can access these items). Attaching a photo of the github footer for reference: 

![New-DAT-GithubFooter](https://github.com/gc-da11yn/gc-da11yn.github.io/assets/136023479/0476a27d-3982-4a88-843c-36be1b9442c5)
(Description: A black footer with the github cat logo titled Contribute on Github.com, with 3 links to the right, see the source code on github.com, report an issue on github.com, and join the discussions on github.com.)

We would love to get feedback on the following items:

**"See source codes on github.com"**
- Would "see" be considered non-inclusive and inconsiderate to visually-impaired people who visit the site?
- If yes, are there any suggestions for possible alternatives? Some ideas we had:
1. Check
2. Learn more about
3. Access 

**"Report an issue on github.com"**
- Simplifying the sentence to a 6th grade reading level by changing to: "Tell us about an issue" - is there any feedback for this change?

**"Join the discussions on github.com"** 
- Currently a college reading level - are there any ideas for simpler language?

**"on github.com"**
For screen reader users, they would hear "on github.com" 4 times. Is that considered redundant or do the links provide more clarity to where they will be taken to after selecting the link? 

Appreciate any feedback and thank you for taking the time to help!

**Comment by martynka** [comment]
From someone who needs clear and concise language I'd like to propose a change. 
**From:**
Contribute on Github.com 
"See the Source Code on Github.com"    "Report an Issue on Github.com"   "Join the discussions on Github.com" 
**To:**
"Contribute to the project on Github"  
"Repositories (Code)"    "Issues"   "Discussions"  

**Notes:**
1. The proposed new link texts would match the tabs on the landing pages
2. Users would greatly benefit if the the landing pages for all 3 links have headings
3. The "Report an Issue" link [[See the source code](https://github.com/gc-da11yn/gc-da11yn.github.io)], takes me to this page: https://github.com/gc-da11yn/gc-da11yn.github.io/issues/new/choose, where the "Issues" tab is emphasized with the underlined.  However, the "Issues" tab link itself takes me to another page. This confuses me. The "Report an Issue" link should point to the "Issues" landing page: https://github.com/gc-da11yn/gc-da11yn.github.io/issues
4. The current link "Report an Issue" landing page is confusing. I don't know how to start, where is the new issue button? This is also the reason why the link should point to the ".../issues" page and not the ".../issues/new/choose" page.

**Reply by shawnthompson** [reply]
@martynka I don't think the "Report an issue" link should take you to a place where you see the issues already created. If you want to report an issue, it should take you to a page that allows you to report a new issue, which is where it takes you. Unfortunately we can't add anymore text to that page because Github takes care of the page and just lists the templates we've created.

There might be a way to customize that chooser page using the [Github API](https://docs.github.com/en/rest?apiVersion=2022-11-28) but I would need someone to do the research on how to do that. 

We want people to report issues they are finding on the site and not just view the list of all the issues I haven't fixed yet. 😢 

If you still think that link should list all the open issues page, then we should be labeling it **View Issues**. But if a user wants to create a new issue they will still land up in the original page that we are linking to and finding themselves in the same situation you stated before. 

This might be something we want to open up an issue with Github for there "Chooser" pages. Explaining that some people might be confused.

**Comment by shawnthompson** [comment]
Here is the new contribute section layout proposed:

![image](https://github.com/gc-da11yn/gc-da11yn.github.io/assets/2575266/bd30ef8a-bfa3-43e6-bc6f-e01bce0b179b)

Alt: White text on black background with Contribute to this project on Github.com (with Github icon) as a heading and the following links, Source code (open and closed square bracket icon), Issues (round circle with dot in the middle icon) and Discussions (chat bubbles icon).

This does look nice but (I hate to say but) I do have a few concerns.

1. As stated ins my comments previous to this one, the issues link, I think, should point to report an issue and not list the issues currently, therefore I would say it should say "[Report an issue or create an issue](https://github.com/gc-da11yn/gc-da11yn.github.io/issues/new/choose)", something more descriptive to the action, some might fail this under [WCAG 2.1 - 2.4.4: Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html) because the link does not describe exactly where it's going.
    - If you really mean it to go to list the "[issues](https://github.com/gc-da11yn/gc-da11yn.github.io/issues)", than we can leave the text as is, but then I would think it would be more difficult for someone to figure out how to report a problem they find who doesn't already know Github. 
2. We should be using the font awesome icons for the icons before the text. 
These are the ones I would suggest (they aren't the exact same as the ones on Github):
    - [Source code](https://fontawesome.com/v5/icons/code?f=classic&s=solid)
    - [Issues](https://fontawesome.com/v5/icons/dot-circle?f=classic&s=regular)
    - [Discussions](https://fontawesome.com/v5/icons/comments?f=classic&s=solid)

Attn: @emmatseng-ux

**Reply by oyinR** [reply]
@shawnthompson how does a person report an issue/problem in Github? And for Font awesome, do we have a pro license to access the all the icons in their library?

**Reply by shawnthompson** [reply]
@oyinR we are using version 5 of the free account for [Font Awesome](https://fontawesome.com/v5/docs).

To report an issue, you would got to the [issues](https://github.com/gc-da11yn/gc-da11yn.github.io/issues) of the project, then click on the [new issue](https://github.com/gc-da11yn/gc-da11yn.github.io/issues/new/choose) button, which then send you to the same page we are currently linking on the site.

**Reply by emmatseng-ux** [reply]
Hey Shawn, the icons you selected look good on my end, the intention of the mock up was to select options that were similar to GitHub to create greater alignment for users navigating between the DAT footer and the GitHub page. The alternatives in FontAwesome does just that, thanks for linking them!

As for the issues page, the intention was to have the link from the footer directly to the issues page in GitHub, not the link to start creating a ticket. During our exploration of the DAT site, we realized that if you didn't have an account, the report an issues link would take you to an account creation/login page rather than the report an issue page in GitHub. Compared to the other 2 links, source code and discussion, you would not be able to explore the GitHub page before signing up for an account through report an issues. As a first time user or non-user of GitHub, this can feel jarring and trapped, feeling forced to sign up for a page they do not know much about. I also don't think it's a bad idea to allow users to explore the existing Issues, understand the various categories that exist, examples of past issues, etc before posting an entry themselves. It's an easier transition to GitHub from my perspective. 

Let me know what you think @shawnthompson! Thanks.

**Reply by shawnthompson** [reply]
@emmatseng-ux that makes sense about it being confusing if you are not logged in. I never realized that because I'm **always** logged in, some might say too much. 

I still think the issues page might be a little much for non Github members. Github is intense for first time users.

What are your thoughts of adding a short disclaimer paragraph / text in that contribute sections about needing a Github account to contribute to this project?

I eventually want to add guides to how to contribute depending on the different roles but currently I do not have the bandwidth to create them.

The W3C is currently going through the same problem with their [WCAG](https://github.com/w3c/wcag) repository and are looking to create guides on how to contribute on Github for non-webdev users. I'd like to align with them and maybe use their guides in the end. Also we could contribute back by translating for them. 

The only guide we have is the [gc-da11yn.github.io/.github/DEVELOPMENT.md](https://github.com/gc-da11yn/gc-da11yn.github.io/blob/main/.github/DEVELOPMENT.md) guide we are hosting on Github, which I'd like to eventually move to an actual webpage on the site.

**Reply by emmatseng-ux** [reply]
@shawnthompson, I met with @oyinR to discuss some of these items you raised, below is our response:

**1. Issues tab in footer**
We agree GitHub for first time users can be intense. For the issues page in particular, we still think the footer should bring them to the general page as opposed to the page to create a new issue. The 3 links in the footer helps to progressively disclose the page to users. By introducing them to each segment of the page, it would help make the overall experience less daunting. 

This pattern of navigation we are proposing for the Issues page, letting users explore the general page before taking an action like creating a post, is actually mirroring what's happening in the Discussions page when selected from the footer currently. This match is good for consistency and what users may expect moving from 1 link to another, navigating all pages similarly. 

**2. adding a disclaimer/paragraph about GitHub account**
At the moment, we don't think this will be necessary. This is because some users have a high aversion to creating accounts, especially to websites they are unfamiliar with. Having this message displayed front and centre may deter them from entering the site altogether, meaning access to important updates and information may be lost. This may also make them feel trapped/forced to create an account when they really do not want to. Additionally, this doesn't feel like working in the open, one of the working principles of the group if people are blocked from access due to the issue of accounts. 

**3. Guides for how to use GitHub**
The guides can be something we explore down the road once our bandwidth is freed up and if we can get feedback on the usability and accessibility of the site post launch. For this current launch, our capacity to support the development of this portion of the project is limited. 

Thanks for your reply as always and let us know what you think!

**Reply by shawnthompson** [reply]
@emmatseng-ux , sounds good I'll start making the changes. 

As for the guides I was just putting the idea out there / here's a to do item if anyone was interested in taking it on.

**Reply by shawnthompson** [reply]
@emmatseng-ux What would issues be in French?

**Reply by shawnthompson** [reply]
This is what we get in mobile:

![image](https://github.com/gc-da11yn/gc-da11yn.github.io/assets/2575266/e4e8ea99-c6e0-4da0-b881-540cf56bafa9)
![image](https://github.com/gc-da11yn/gc-da11yn.github.io/assets/2575266/c414cf7c-c7e6-4a34-bb01-6d75b283faa1)

Keeping in mind "Enjeu" is a direct translation to "Issues"

**Reply by emmatseng-ux** [reply]
Thanks for sharing the staging link @shawnthompson! We consulted Melissa and Laurent on our team for help with French translations. As GitHub is a uni-lingual site, per Official Languages, we will have to indicate that in the link. Posting the team's suggestions below:

1. Problèmes (en anglais seulement)
2. Problèmes (Issues) (en anglais seulement)
3. Issues (problèmes) (en anglais seulement) 

I personally am leaning towards Option #2 because the English equivalent is also noted there which will further help with navigation. Wondering for stylistic purposes, can we format the space between the title and anglais seulement with a vertical bar and remove the brackets? 
- Ex: Problèmes (Issues) | en anglais seulement

Side note: the English version for Issues is missing an 's' plural, in staging.

**Reply by shawnthompson** [reply]
@emmatseng-ux, I've added the "s" to the links text for "Issues".

I like the 2 option also although I'd like to keep it formatted the same as elsewhere on our site with the <span lang="fr">_(en anglais seulement)_</span> text. In some cases we are even using the  <span lang="fr">_(**Disponible** en anglais seulement)_</span>. I'd like to stay consistent throughout the site with the format now and open an issue / discussion to talk about it in more detail. All through the Government of Canada websites they are using brackets "()" around the extra information to links so I believe we should be consistent to the rest of the sites for now. Eventually pushing for change to the over all GC web.

I did a [code search results for "en anglais seulement"](https://github.com/search?q=repo%3Agc-da11yn%2Fgc-da11yn.github.io+%22en+anglais+seulement%22&type=code).

The Canada.ca Style guide has the following guidance. The accessibility community is not a fan of their version because it places the _(French only)_ text outside of the actual link text and we feel it should be inside the link text as it's important information needed to understand the link text. So I really don't want to follow their guidance but wanted to acknowledge that we know about it.

See section [7.4 Linking to non-government content](https://www.canada.ca/en/treasury-board-secretariat/services/government-communications/canada-content-style-guide.html#wp7-4) for more details.

I will start a [Best Practices Open Discussions](https://github.com/gc-da11yn/gc-da11yn.github.io/discussions/categories/best-practices-open-discussions) about why we should change the format and come to an agreement to change it.

**Reply by shawnthompson** [reply]
Discussion about links available in other language started at https://github.com/gc-da11yn/gc-da11yn.github.io/discussions/311

**Comment by shawnthompson** [comment]
Here is a staged version of the changes:

[Development version: Digital Accessibility Toolkit - Digital Accessibility Toolkit / Sharing space](https://deploy-preview-308--a11ycanada.netlify.app/en/)

Keep in mind that the Netlify text only shows up on the staging environment. It's a condition to use Netlify's Open source plan, which we are not using on the live site.

**Reply by oyinR** [reply]
@shawnthompson will all dev changes including the homepage designs show up in this staging environment? Also when would be a good time to review designs in staging?

**Reply by shawnthompson** [reply]
The whole site is a "staging" environment so we will be updating as much as needed and not waiting on multiple changes. The aim is to "launch" in December with mass promotions about the site. We are keeping this as agile as possible. We don't have the resources to setup a staging site for every change and group them together for one big release.

**Reply by oyinR** [reply]
You shared a [development version of the site](https://deploy-preview-308--a11ycanada.netlify.app/en/) which has a different URL to the [DAT site](https://a11y.canada.ca/en/). Which one should we refer to for updates?  Also wondering when will the homepage designs be updated? Is there timeline/roadmap we can refer to or would you be informing us when the designs have been updated?

**Reply by shawnthompson** [reply]
Those development versions are setup to review every change we make, also know as [pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests). The one above if for the change in the footer. 

As for the homepage change, I'm looking to have that done in the following week or two. With holidays and being the only person making changes I have a backlog of changes to make.

The homepage isn't a priority for me at the moment. 

On a good note, @acev0010 has joined my team of one to help me with the backlog of changes so it should move faster once I get her up to speed.

**Reply by oyinR** [reply]
Great, thanks Shawn! I'm so happy you have more support now! And @acev0010, welcome to the team!
Julie shared a list of changes you will be making in the next  few weeks. I'll set up a touchpoint for us, if the time/day doesn't work please propose a new date!
